pkg/logger/logger.go

```diff
@@ -13,12 +13,20 @@ import (
 	logf "sigs.k8s.io/controller-runtime/pkg/runtime/log"
 )
 
+var outputTo io.Writer = os.Stderr
+
 // Logger shared logger struct
 type Logger struct {
 	Logger        logr.Logger
 	SugaredLogger *zap.SugaredLogger
 }
 
+// GetLoggerWithOut returns a custom named logger to out in a specific writer
+func GetLoggerWithOut(name string, w io.Writer) *zap.SugaredLogger {
+	outputTo = w
```

Comment:
Is this going to affect the result of future invocation of function `GetLogger`, which could make `GetLogger` unpredictable?

Reply:
Absolutely. Each `GetLogger*` will have its own output since they create a new logger each time is called. Calling `GetLogger` will force the output to the `os.StdErr`. This one has an option to output to another `writer`. Is used during CLI unit tests to verify the output of each command and assert accordingly.

Reply:
Then, I think we should make sure calling `GetLogger`  **directly**  always forces the output to `os.StdErr` (e.g. reset `outputTo` to `os.StdErr` before exiting `GetLoggerWithOut` ). Otherwise we won't  be able to know which output it uses after `GetLoggerWithOut` is called. Another option maybe we can provide a function to change the `outputTo` directly.

Reply:
Done!